perma_web/perma/tasks.py

```diff
@@ -869,12 +869,18 @@ def delete_from_internet_archive(self, link_guid):
 
 @shared_task()
 def upload_all_to_internet_archive():
-    # find all links created 48-24 hours ago
-    # include timezone
-    start_date = timezone.now() - timedelta(days=2)
-    end_date   = timezone.now() - timedelta(days=1)
+    """
+        Links to upload:
+        - links that have not started to upload
+        - links that have failed to upload but not failed permanently (those should not exist, but doesn't hurt to look)
+        - links that have previously been deleted from internet archive (is_private=True) but have since been made public
+        - only links that are older than 24 hours
+    """
+    links = Link.objects.filter((Q(internet_archive_upload_status='not_started') |
+                                Q(internet_archive_upload_status='failed') |
+                                Q(internet_archive_upload_status='deleted', is_private=False)) &
```

Comment:
We delete previously public links that are going private from Internet Archive https://github.com/harvard-lil/perma/blob/develop/perma_web/api/resources.py#L632

We should upload links previously private that are going public TO Internet Archive